src/service/resources-impl.js

```diff
@@ -362,11 +362,21 @@ export class Resources {
       this.viewport_.ensureReadyForElements();
     }
 
-    // Create and add the resource.
-    const resource = new Resource((++this.resourceIdCounter_), element, this);
+    // First check if the resource is being reparented and if it requires
+    // reconstruction. Only already built elements are eligible.
+    let resource = Resource.forElementOptional(element);
+    if (resource &&
+        resource.getState() != ResourceState.NOT_BUILT &&
```

Comment:
So, I'm kind of leading this from a current state, which is `reconstructWhenReparented  = true` for all cases. The `!built` case is ultimately doesn't matter - there's no state so we are not really winning or loosing anything with that. Also, `Resource` class already knows how to work correctly with "built" elements (see https://github.com/ampproject/amphtml/blob/master/src/service/resource.js#L147).